### PR TITLE
Force a string `true` in the update_shared_storages to match the expectations of  _mount_unmount

### PR DESCRIPTION
**Pull request by dreambeyondorange**
… in the efs resource

Updates to a cluster result in boolean values for the iam_authorization and efs_encryption_in_transit objects.

### Tests
* Tested the change locally with an update to a cluster with an EFS fs with encryption in transit.  Adding the boolean check allows the cookbook to use TLS for the fs.

### Checklist
- Make sure you are pointing to **the right branch**.
- If you're creating a patch for a branch other than `develop` add the branch name as prefix in the PR title (e\.g\. `[release-3.6]`).
- Check all commits' messages are clear, describing what and why vs how.
- Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cookbooks/aws-parallelcluster-environment/resources/efs/partial/_mount_umount.rb [hunk 1]
```diff
@@ -40,9 +40,9 @@
 
     # See reference of mount options: https://docs.aws.amazon.com/efs/latest/ug/automount-with-efs-mount-helper.html
     mount_options = "_netdev,noresvport"
-    if efs_encryption_in_transit == "true"
+    if efs_encryption_in_transit == "true" || efs_encryption_in_transit == true
```
Comment (gmarciani) [inline cookbooks/aws-parallelcluster-environment/resources/efs/partial/_mount_umount.rb]: We introduced encryption and iam authz in 3.4.0. Does this bug means that it has never worked?
What determines if the variable uses the string or the boolean?
Instead of making a double check string&bool I suggest two alternatives:
1. (enough) convert the variable to boolean and check only for the boolean value
2. (the best) fix the code at the source of the issue, so that the value can only be a string or a boolean, and check just for one of them.

Also, I imagine we may have this type of situation also in other places. I suggest to create a task to check cookbook wide to prevent future issues.

Comment (dreambeyondorange) [reply cookbooks/aws-parallelcluster-environment/resources/efs/partial/_mount_umount.rb]: I think the problem is just in update, since the dna.json doesn't have booleans, just strings and the update generates the data.  It seems adding an EFS fs with encryption in transit wasn't tested, and is an edge case since no one has reported an issue until now.

Comment (dreambeyondorange) [reply cookbooks/aws-parallelcluster-environment/resources/efs/partial/_mount_umount.rb]: Either converting a string to a boolean or checking for the string and the boolean seems the same to me.  Why is the former better than the latter?  Actually I can answer this myself ahaha, but the code generally doesn't allow us to force typing, and that is how we end up in these situations.  We can convert the boolean that is generated from the yaml loading to a string.  Unless we document these types of patterns, this will trip up engineers in the future as well.  Since how would they know what the variable type should be?

Comment (dreambeyondorange) [reply cookbooks/aws-parallelcluster-environment/resources/efs/partial/_mount_umount.rb]: But considering we use the data from dna.json throughout the codebase, we should force these to be strings to match that pattern

File: cookbooks/aws-parallelcluster-environment/recipes/config/update_shared_storages.rb [hunk 2]
```diff
@@ -83,8 +83,17 @@ def get_efs(action)
         next unless not_in_shared_storages_mapping["efs"].nil? || !not_in_shared_storages_mapping["efs"].include?(storage)
         shared_dir_array.push(storage["mount_dir"])
         efs_fs_id_array.push(storage["efs_fs_id"])
-        efs_encryption_in_transit_array.push(storage["efs_encryption_in_transit"])
-        efs_iam_authorization_array.push(storage["efs_iam_authorization"])
+        # The EFS resource expects strings for these attributes, not booleans
+        if storage["efs_encryption_in_transit"]
```
Comment (gmarciani) [inline cookbooks/aws-parallelcluster-environment/recipes/config/update_shared_storages.rb]: What about avoiding if..else block as follows?

```
efs_encryption_in_transit_array.push(String(storage["efs_encryption_in_transit"]))
```

Comment (dreambeyondorange) [reply cookbooks/aws-parallelcluster-environment/recipes/config/update_shared_storages.rb]: oh thanks this is great!

Comment (dreambeyondorange) [reply cookbooks/aws-parallelcluster-environment/recipes/config/update_shared_storages.rb:87]: Although I'm not sure that the object is guaranteed to exist and String cannot handle empty input

**Review by gmarciani — APPROVED** [review]
